shared/clients/gemini_interaction.py

```
@@ -289,6 +293,16 @@ async def stream_interaction_result(
                     )
 
+            except httpx.TimeoutException:
+                logger.info(
+                    "[GEMINI_CLIENT][STREAM] Read timeout after %.0fs: id=%s, events_received=%d",
+                    stream_timeout or 0,
+                    interaction_id,
+                    event_count,
+                )
+                raise GeminiInteractionError(
+                    f"Stream read timeout after {stream_timeout}s",
+                )
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)